test/pytest/test_packages.py

```diff
@@ -239,3 +239,49 @@ def test_translation(pkgdir):
     assert b'eins\n' in contents
     assert b'zwo\n' in contents
     assert b'zwei\n' not in contents
+
+
+def test_filename_mangling(pkgdir):
+    make_package(pkgdir, 'one')
+
+    # test various filename variations
+    (pkgdir / 'one' / 'one.js').write_text('this is one.js')
+    (pkgdir / 'one' / 'two.js.gz').write_text('this is two.js')
+    (pkgdir / 'one' / 'three.min.js.gz').write_text('this is three.js')
```

Comment:
These two are a bit odd, as they aren't actually gzipped. But I figure we are not testing `Transport-Encoding:` etc. here, and pretend that these are just opaque binary blobs?